src/Poseidon/GenotypeData.hs

```diff
@@ -256,18 +259,30 @@ recodeAlleles consensusSnpEntry snpEntry genoLine = do
     flipGeno HomAlt = HomRef
     flipGeno g      = g
 
-printSNPCopyProgress :: (MonadIO m) => Pipe a a m ()
-printSNPCopyProgress = do
+printSNPCopyProgress :: (MonadIO m) => LogEnv -> UTCTime -> Pipe a a m ()
+printSNPCopyProgress logEnv startTime = do
     counterRef <- liftIO $ newIORef (0 :: Int)
     for cat $ \val -> do
         n <- liftIO $ readIORef counterRef
-        when (n `rem` 1000 == 0) $ do
-            liftIO $ hClearLine stderr
-            liftIO $ hSetCursorColumn stderr 0
-            liftIO $ hPutStr stderr ("> " ++ show n ++ " ")
-            liftIO $ hFlush stderr
+        currentTime <- liftIO getCurrentTime
+        logWithEnv logEnv $ logProgress n (diffUTCTime currentTime startTime)
         liftIO $ modifyIORef counterRef (+1)
         yield val
+    where
+        logProgress :: Int -> NominalDiffTime -> PoseidonLogIO ()
+        logProgress c t
+            | c >=  1000000 &&                c `rem` 1000000 == 0 = logInfo $ "SNPs: " ++ pad (show c) ++ " - " ++ prettyTime t
+            | c >=  100000  && c <= 500000 && c `rem` 100000  == 0 = logInfo $ "SNPs: " ++ pad (show c) ++ " - " ++ prettyTime t
+            | c >=  10000   && c <= 50000  && c `rem` 10000   == 0 = logInfo $ "SNPs: " ++ pad (show c) ++ " - " ++ prettyTime t
+            |                  c <= 5000   && c `rem` 1000    == 0 = logInfo $ "SNPs: " ++ pad (show c) ++ " - " ++ prettyTime t
+            | otherwise = return ()
+        pad :: String -> String
+        pad s
+            | length s < 9  = replicate (9 - length s) ' ' ++ s
+            | otherwise     = s
+        prettyTime :: NominalDiffTime -> String
+        prettyTime = formatTime defaultTimeLocale "%H:%M:%S" . posixSecondsToUTCTime
```

Comment:
Isn't there a simpler way to do this time-wise logging? I would think that, if we want a time-stamp in our logging, we should do that higher up in our log-function? Or do you think it's important to have time-stamps only here in the SNP progress, but not in all other logging places? I don't really see why this distinction is necessary... either time-stamps are important for debugging logging-information, or not, but why would they be interesting in some cases but not others?

Reply:
> I now understand that the time-stamps after the SNP count denotes time passed since the start of the download, not timestamp of the log, which seems less redundant than I thought above. We can keep that, I guess. I think if I had implemented that, though, I wouldn't have put this in, since the user in principle has a way to deal with this using --logMode ServerLog (which I think we should rename or redesign to make it easier to print time stamps).

I thought it may be useful for the user to quickly estimate how long a given process will still take. To do that "right", we would need an estimate of the number of total SNPs to be merged (as discussed above). Then we can give a proper ETA.

Reply:
I implemented a human-optimized time display now:
```
[Info]    SNPs:    420000    55s
[Info]    SNPs:    430000    56s
[Info]    SNPs:    440000    58s
[Info]    SNPs:    450000    59s
[Info]    SNPs:    460000    1m 0s
[Info]    SNPs:    470000    1m 2s
[Info]    SNPs:    480000    1m 3s
[Info]    SNPs:    490000    1m 4s
```
I think that adds enough value to keep it around independent of the timestamp one can get through the logging infrastructure.